src/dms/core/EdFi.DataManagementService.Core/ApiSchema/ApiSchemaFileLoader.cs

```diff
@@ -40,168 +40,57 @@ private static JsonNode LoadFromAssembly(string resourceName, Assembly assembly)
         {
-            throw new InvalidOperationException($"Access denied to directory '{directoryPath}'", ex);
+            throw new InvalidOperationException("ApiSchemaPath is not configured.");
         }
+        JsonNode coreApiSchemaNode = new JsonObject();
+        JsonNode[] extensionApiSchemaNodes = Array.Empty<JsonNode>();
 
-        return fileContents;
-    }
+        var projectDirectory = Path.GetFullPath(Path.Combine(AppDomain.CurrentDomain.BaseDirectory, "../../../"));
+        var relativeToProject = Path.GetFullPath(Path.Combine(projectDirectory, appSettings.Value.ApiSchemaPath));
 
-    /// <summary>
-    /// Returns core and extension ApiSchema JsonNodes
-    /// </summary>
-    private readonly Lazy<ApiSchemaNodes> _apiSchemaNodes = new(() =>
-    {
-        if (appSettings.Value.UseApiSchemaPath)
+        if (!Directory.Exists(relativeToProject))
         {
-            string apiSchemaPath =
-                appSettings.Value.ApiSchemaPath
-                ?? throw new InvalidOperationException("No ApiSchemaPath configuration is set");
+            throw new InvalidOperationException("relativeToProject is not exists.");
+        }
 
-            if (!Directory.Exists(apiSchemaPath))
-            {
-                throw new InvalidOperationException($"The directory {apiSchemaPath} does not exist.");
-            }
+        string apiSchemaPath = Path.GetFullPath(relativeToProject);
+        var assemblies = Directory.GetFiles(apiSchemaPath, "*.dll", SearchOption.AllDirectories);
+        var apiSchemaAssemblyLoadContext = new ApiSchemaAssemblyLoadContext();
+        foreach (var assemblyPath in assemblies)
+        {
+            var assembly = apiSchemaAssemblyLoadContext.LoadFromAssemblyPath(assemblyPath);
 
-            List<JsonNode> apiSchemaNodes = ReadApiSchemaFiles(apiSchemaPath);
+            var manifestResourceNames = assembly.GetManifestResourceNames();
 
-            JsonNode coreApiSchemaNode = apiSchemaNodes.First(node =>
-                !node.SelectRequiredNodeFromPathAs<bool>("$.projectSchema.isExtensionProject", _logger)
+            var coreSchemaResourceName = Array.Find(
+                manifestResourceNames,
+                str => str.EndsWith("ApiSchema.json")
             );
 
-            JsonNode[] extensionApiSchemaNodes = apiSchemaNodes
-                .Where(node =>
-                    node.SelectRequiredNodeFromPathAs<bool>("$.projectSchema.isExtensionProject", _logger)
-                )
-                .ToArray();
+            var extensionSchemaResourceName = Array.Find(
+                manifestResourceNames,
+                str => str.Contains(".ApiSchema-") && str.EndsWith("EXTENSION.json")
+            );
 
-            return new(coreApiSchemaNode, extensionApiSchemaNodes);
-        }
-        else
-        {
-            Assembly coreAssembly =
-                Assembly.GetAssembly(typeof(DataStandard52.ApiSchema.Marker))
```

Comment:
I still want to bundle the DS and TPDM for quick start like before. What I don't want is anything Sample or Homograph anywhere. Those are test extensions. They are still in the packages list. Search for them and remove them.

Reply:
I am bundling only DS and TPDM for quick start